auth/tokenizer/asymmetric/tokenizer.go

```diff
@@ -0,0 +1,252 @@
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+const patPrefix = "pat"
+
+var (
+	errLoadingPrivateKey = errors.New("failed to load private key")
+	errInvalidKeySize    = errors.New("invalid ED25519 key size")
+	errParsingPrivateKey = errors.New("failed to parse private key")
+	errInvalidKeyType    = errors.New("private key is not ED25519")
+	errNoValidPublicKeys = errors.New("no valid public keys available")
+	errNoActiveKey       = errors.New("active key not loaded")
+)
+
+type keyPair struct {
+	id         string
+	privateKey jwk.Key
+	publicKey  jwk.Key
+}
+
+type tokenizer struct {
+	activeKey   *keyPair
+	retiringKey *keyPair // Optional, for key rotation grace period
+}
+
+var _ auth.Tokenizer = (*tokenizer)(nil)
+
+// NewTokenizer creates a new asymmetric tokenizer with active and optionally retiring keys.
+// activeKeyPath is required. retiringKeyPath is optional (can be empty string).
+// If retiringKeyPath is provided but the file doesn't exist or is invalid, a warning is logged
+// but the tokenizer is still created with just the active key.
+// Key IDs are derived from filenames to ensure consistency across multiple service instances.
+func NewTokenizer(activeKeyPath, retiringKeyPath string, idProvider supermq.IDProvider, logger *slog.Logger) (auth.Tokenizer, error) {
+	activeKID := keyIDFromPath(activeKeyPath)
+
+	activePrivateJwk, activePublicJwk, err := loadKeyPair(activeKeyPath, activeKID)
+	if err != nil {
+		return nil, err
+	}
+
+	mgr := &tokenizer{
+		activeKey: &keyPair{
+			id:         activeKID,
+			privateKey: activePrivateJwk,
+			publicKey:  activePublicJwk,
+		},
+	}
+
+	if retiringKeyPath != "" {
+		retiringKID := keyIDFromPath(retiringKeyPath)
+
+		retiringPrivateJwk, retiringPublicJwk, err := loadKeyPair(retiringKeyPath, retiringKID)
+		if err != nil {
+			logger.Warn("failed to load retiring key, continuing without it", slog.Any("error", err))
+			return mgr, nil
+		}
+
+		mgr.retiringKey = &keyPair{
+			id:         retiringKID,
+			privateKey: retiringPrivateJwk,
+			publicKey:  retiringPublicJwk,
+		}
+		logger.Info("loaded retiring key for rotation grace period", slog.String("key_id", retiringKID))
+	}
+
+	return mgr, nil
+}
+
+func keyIDFromPath(path string) string {
+	base := filepath.Base(path)
+	ext := filepath.Ext(base)
+	return strings.TrimSuffix(base, ext)
+}
+
+func (km *tokenizer) Issue(key auth.Key) (string, error) {
+	if km.activeKey == nil {
+		return "", errNoActiveKey
+	}
+
+	builder := jwt.NewBuilder()
+	builder.
+		Issuer(smqjwt.IssuerName).
+		IssuedAt(key.IssuedAt).
+		Claim(smqjwt.TokenType, key.Type).
+		Expiration(key.ExpiresAt).
+		Claim(smqjwt.RoleField, key.Role).
+		Claim(smqjwt.VerifiedField, key.Verified)
+
+	if key.Subject != "" {
+		builder.Subject(key.Subject)
+	}
+	if key.ID != "" {
+		builder.JwtID(key.ID)
+	}
+
+	tkn, err := builder.Build()
+	if err != nil {
+		return "", err
+	}
```

Comment:
Can we abstract this function to common , because this is used between asymmetric and symmetric ?   
   
We can build a abstract object, that can be used between asymmetric and symmetric ?